app/navigation/TransportNavigator.scala

```diff
@@ -35,10 +41,45 @@ trait TransportNavigatorProvider {
 }
 
 class TransportNavigator(override val mode: Mode)(implicit override val appConfig: FrontendAppConfig, override val phaseConfig: PhaseConfig)
-    extends UserAnswersNavigator {
+    extends UserAnswersNavigator
+    with CustomNavigator {
 
   override type T = TransportDomain
 
   implicit override val reader: UserAnswersReader[TransportDomain] =
     TransportDomain.userAnswersReader
+
+  override def nextPage(userAnswers: UserAnswers, currentPage: Option[Page]) = currentPage match {
+    case Some(page) => navigateToNextPage(page, userAnswers, mode)
+    case None       => super.nextPage(userAnswers, currentPage)
+  }
+
+  override protected def normalRoutes(mode: Mode) = {
+    case AddAdditionalReferenceYesNoPage      => ua => additionalRefYesNoRoute(ua, mode)
+    case AdditionalReferencesSection          => ua => AddAdditionalInformationYesNoPage.route(ua, mode)
+    case AddAdditionalInformationYesNoPage    => ua => additionalInfoYesNoRoute(ua, mode)
+    case AdditionalInformationTypePage(index) => ua => AddCommentsYesNoPage(index).route(ua, mode)
+    case AddCommentsYesNoPage(index)          => ua => addCommentYesNoRoute(ua, index, mode)
+    case AdditionalInformationTextPage(_) =>
+      ua => Some(controllers.additionalInformation.routes.AddAnotherAdditionalInformationController.onPageLoad(ua.lrn, mode))
+    case AdditionalInformationListSection => ua => Some(controllers.routes.TransportAnswersController.onPageLoad(ua.lrn))
+  }
+
+  private def additionalRefYesNoRoute(userAnswers: UserAnswers, mode: Mode): Option[Call] =
+    userAnswers.get(AddAdditionalReferenceYesNoPage) flatMap {
+      case true  => ???
+      case false => AddAdditionalInformationYesNoPage.route(userAnswers, mode)
+    }
+
+  private def additionalInfoYesNoRoute(userAnswers: UserAnswers, mode: Mode): Option[Call] =
+    userAnswers.get(AddAdditionalInformationYesNoPage) flatMap {
+      case true  => AdditionalInformationTypePage(Index(0)).route(userAnswers, mode)
+      case false => Some(controllers.routes.TransportAnswersController.onPageLoad(userAnswers.lrn))
+    }
+
+  private def addCommentYesNoRoute(userAnswers: UserAnswers, index: Index, mode: Mode): Option[Call] =
+    userAnswers.get(AddCommentsYesNoPage(index)) flatMap {
+      case true  => AdditionalInformationTextPage(index).route(userAnswers, mode)
+      case false => Some(controllers.additionalInformation.routes.AddAnotherAdditionalInformationController.onPageLoad(userAnswers.lrn, mode))
+    }
```

Comment:
@karahanozturk this isn't how the navigation works in departures. This is defined by the user answers readers in the journey domain models. Happy to pair with you if needed.